datafusion/optimizer/Cargo.toml

```diff
@@ -45,6 +45,7 @@ datafusion-expr = { path = "../expr", version = "15.0.0" }
 datafusion-physical-expr = { path = "../physical-expr", version = "15.0.0" }
 hashbrown = { version = "0.13", features = ["raw"] }
 log = "^0.4"
+regex-syntax = "0.6.28"
```

Comment:
This is technically not a new dependency (as datafusion-physical-expr already depends on `regex` which depends on `regex-syntax`). However, regex is an optional dependency whereas this is not optional.

I think it would be better to move this dependency into `datafusion-physical-expr` so regex can still be optional. I will try to do so over the next few days